### PR TITLE
Change flatten

### PR DESCRIPTION
**Pull request by TimurTarasenko**
Change on flat_map with flatten because flat_map better  is no need to
create an intermediate array.

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
CC @reitermarkus for tboughts.

**Comment by MikeMcQuaid** [comment]
@TimurTarasenko You've broken `brew tests`.

**Comment by jawshooah** [comment]
`arr.flat_map` is equivalent to `arr.map.flatten`, *not* `arr.flatten.map`.

File: Library/Homebrew/utils/formatter.rb [hunk 1]
```diff
@@ -51,7 +51,7 @@ def prefix(prefix, string, color)
   private_class_method :prefix
 
   def columns(*objects, gap_size: 2)
-    objects = objects.flatten.map(&:to_s)
+    objects = objects.flat_map(&:to_s)
```
Comment (jawshooah) [inline Library/Homebrew/utils/formatter.rb]: `arr.flat_map` is equivalent to `arr.map.flatten`, *not* `arr.flatten.map`, unfortunately.

File: Library/Homebrew/vendor/macho/macho/fat_file.rb [hunk 2]
```diff
@@ -161,7 +161,7 @@ def linked_dylibs
       # Individual architectures in a fat binary can link to different subsets
       # of libraries, but at this point we want to have the full picture, i.e.
       # the union of all libraries used by all architectures.
-      machos.map(&:linked_dylibs).flatten.uniq
+      machos.flat_map(&:linked_dylibs).uniq
```
Comment (jawshooah) [inline Library/Homebrew/vendor/macho/macho/fat_file.rb]: This, on the other hand, is fine :+1:

**Comment by MikeMcQuaid** [comment]
Closing due to broken tests.

**Pull request closed without merging**